authorize_helper.go

```diff
@@ -30,24 +30,6 @@ import (
 	"github.com/pkg/errors"
 )
 
-// GetRedirectURIFromRequestValues extracts the redirect_uri from values but does not do any sort of validation.
-//
-// Considered specifications
-// * https://tools.ietf.org/html/rfc6749#section-3.1.2
-//   The endpoint URI MAY include an
-//   "application/x-www-form-urlencoded" formatted (per Appendix B) query
-//   component ([RFC3986] Section 3.4), which MUST be retained when adding
-//   additional query parameters.
-func GetRedirectURIFromRequestValues(values url.Values) (string, error) {
```

Comment:
No, as I mentioned. You should still validate that the url is correct. So just parse the URL, check error return value, and this is it.

Reply:
Sorry, I realized that this function does not have to validate anything, it is written in its description and redirect URI validation is done later in `MatchRedirectURIWithClientRedirectURIs`.

I do agree that it could probably just be removed. But let's wait for @aeneasr to confirm.